### PR TITLE
[FLINK-6581] [cli] Correct dynamic property parsing for YARN cli

### PR DESCRIPTION
**Pull request by tillrohrmann**
The YARN cli will now split the dynamic propertie at the first occurrence of
the = sign instead of splitting it at every = sign. That way we support dynamic
properties of the form -yDenv.java.opts="-DappName=foobar".

### CODE REVIEW

File: flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java [hunk 1]
```diff
@@ -702,9 +702,15 @@ private void logAndSysout(String message) {
...
-				String[] kv = propLine.split("=");
-				if (kv.length >= 2 && kv[0] != null && kv[1] != null && kv[0].length() > 0) {
-					properties.put(kv[0], kv[1]);
+				int firstEquals = propLine.indexOf("=");
+
+				if (firstEquals >= 0) {
+					String key = propLine.substring(0, firstEquals).trim();
+					String value = propLine.substring(firstEquals + 1, propLine.length()).trim();
+
+					if (!key.isEmpty() && !value.isEmpty()) {
```
Comment (zentol) [inline flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java]: We didn't reject empty values before. Not sure if there's a valid use-case for that though, maybe overriding/clearing some variable?

Comment (tillrohrmann) [inline flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java]: True, in order to keep in sync with the old behaviour, I'll remove the check whether a value is empty.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol. I'll address your comment and then merge the PR once Travis gives green light.

**Comment by rmetzger** [comment]
+1

**Comment by johnboy14** [comment]
Thanks for turning this around quickly @tillrohrmann We are experiencing this issue with AWS EMR.  Is it possible to patch the current 1.2.0 version available via EMR?

**Comment by tillrohrmann** [comment]
@johnboy14 I can merge this PR also into the `1.2` branch. This means that it will be included in the next bug fix release `1.2.2`. How quickly it will be included in EMR depends a little bit on the AWS/Apache BigTop. We can try to ping them. But if customers complain, then it's much more efficient.

**Comment by johnboy14** [comment]
When's the next bug fix release scheduled for @tillrohrmann ?

**Comment by tillrohrmann** [comment]
This is not decided yet. Usually we do it depending on the urgency of the fixes and the time since the last bug fix release. Currently, we're working on the `1.3` release which requires most of our time. Thus, I would assume not before mid of next month but this is a community decision.

**Comment by johnboy14** [comment]
At this time I am looking a workaround in the meantime. Is there anyway to override the log4j.properties file for each application deployed to EMR?

**Pull request closed without merging**